cmd/pulumi-test-language/tests.go

```diff
@@ -308,8 +308,10 @@ var languageTests = map[string]languageTest{
 
 					outputs := stack.Outputs
 
-					assert.Len(l, outputs, 4, "expected 4 outputs")
-					assertPropertyMapMember(l, outputs, "empty", resource.NewObjectProperty(resource.PropertyMap{}))
+					assert.True(l, len(outputs) == 3 || len(outputs) == 4, "there are 3 or 4 outputs")
+					if empty, ok := outputs[resource.PropertyKey("empty")]; ok {
+						assert.Equal(l, resource.NewObjectProperty(resource.PropertyMap{}), empty)
+					}
```

Comment:
This is part here is up for discussion: whether want to make it OK that an empty map/list output doesn't register anything or change program-gen to emit empty structures

Reply:
I think given the other languages _do_ register an object here it's more consistent to change Go programgen to fix this.
I'm also not massively keen on the thinking that "null == empty map" especially in terms of stack outputs which are a multi-language boundary layer. If we deserialise stack outputs else where (due to stack references, or sub-program components) I'd rather just deal with null is null, and maps are maps.

Reply:
I opted for generating empty structures now. Found an edge case where `nil` isn't valid for an empty map which is for output invokes that have no inputs; they still accept a struct of args but it is not a pointer.